packages/toolkit/src/view/settings/integrations/Integrations.tsx

```diff
@@ -0,0 +1,205 @@
+"use client";
+
+import * as React from "react";
+import { Integration, IntegrationConnection } from "instill-sdk";
+
+import { Icons, Input, Nullable, Skeleton } from "@instill-ai/design-system";
+
+import {
+  debounce,
+  useInfiniteIntegrationConnections,
+  useInfiniteIntegrations,
+} from "../../../lib";
+import { AvailableIntegration } from "./AvailableIntegration";
+import { ExistingConnection } from "./ExistingConnection";
+import { Section } from "./Section";
+
+export type IntegrationsProps = {
+  namespaceId: Nullable<string>;
+  enableQuery: boolean;
+  accessToken: Nullable<string>;
+};
+
+export const Integrations = (props: IntegrationsProps) => {
+  const { accessToken, enableQuery, namespaceId } = props;
+  const [searchInputValue, setSearchInputValue] =
+    React.useState<Nullable<string>>(null);
+  const debouncedSetSearchValue = React.useMemo(
+    () =>
+      debounce((value: string) => {
+        setSearchInputValue(value);
+      }, 500),
+    [],
+  );
+  const availableIntegrations = useInfiniteIntegrations({
+    accessToken,
+    enabled: enableQuery,
+    filter: searchInputValue ? `qIntegration="${searchInputValue}"` : null,
+  });
+  const integrationConnections = useInfiniteIntegrationConnections({
+    namespaceId,
+    accessToken,
+    enabled: enableQuery,
+    filter: searchInputValue ? `qConnection="${searchInputValue}"` : null,
+  });
+
+  React.useEffect(() => {
+    if (availableIntegrations.data) {
+      if (
+        availableIntegrations.data.pages[
+          availableIntegrations.data.pages.length - 1
+        ]?.nextPageToken
+      ) {
+        availableIntegrations.fetchNextPage();
+      }
+    }
+
+    if (integrationConnections.data) {
+      if (
+        integrationConnections.data.pages[
+          integrationConnections.data.pages.length - 1
+        ]?.nextPageToken
+      ) {
+        integrationConnections.fetchNextPage();
+      }
+    }
+  }, [
+    availableIntegrations.isSuccess,
+    availableIntegrations.data,
+    integrationConnections.isSuccess,
+    integrationConnections.data,
+  ]);
+
+  const availableIntegrationList = React.useMemo(() => {
+    return (
+      availableIntegrations.data?.pages.reduce(
+        (acc: Integration[], page) => acc.concat(page.integrations),
+        [],
+      ) ||
+      //.sort((a, b) => (a.title < b.title ? -1 : a.title > b.title ? 1 : 0))
+      []
+    );
```

Comment:
@orangecoloured Try not to use reduce to concat the array, simply use for loop to make it easier to read